sigpy/linop.py

```diff
@@ -116,7 +120,24 @@ def H(self):
             Linop: adjoint linear operator.
 
         """
-        return self._adjoint_linop()
+        if self.adj is None:
+            self.adj = self._adjoint_linop()
+        return self.adj
+
+    @property
+    def N(self):
+        r"""Return normal linear operator.
+
+        A normal linear operator :math:`A^HA` for
+        a linear operator :math:`A`.
+
+        Returns:
+            Linop: adjoint linear operator.
+
+        """
+        if self.normal is None:
```

Comment:
See above. Maybe call an _normal_linop function, which implements a default normal operator self.H * self.

Users can override _normal_linop to implement their own

Reply:
Sounds good!